lib/friendly_uuid.rb

```diff
@@ -20,12 +20,26 @@ def to_param
   end
 
   module Class
-    def find(id)
-      super(self.expand(id))
+    def find(*ids)
+      super(self.expand(ids))
     end
 
     def expand(short_uuid)
```

Comment:
Thanks for the thoughts. From my testing, there's a bit of nuance in regard to what is returned by vanilla `#find` and `#find_by`:

```ruby
# Passing a non-array as a parameter to #find
Object.find(1).class
=> Object

# Passing a one-element array as a parameter to #find
Object.find([1]).class
=> Array

# Passing multiple values as a non-explicit array as a parameter to #find
Object.find(1, 2).class
=> Array

# Passing an explicit multiple-element array to #find 
Object.find([1, 2]).class
=> Array

# Passing a non-array as a parameter to #find_by
Object.find_by(id: 1).class
=> Object

# Passing a single-element array as a parameter to #find_by
Object.find_by(id: [1]).class
=> Object

# Passing a multiple-element array as a parameter to #find_by
Object.find_by(id: [1, 2]).class
=> Object
```

Due to this, we can't treat

```ruby
User.find("a", "b", "c")
User.find(["a", "b", "c"])
User.find("a")
```

all the same, as the first two return arrays, while the last returns an object. Or am I missing something? (Always a possibility!)

Reply:
Thanks for the response, @glacials, and sorry for the late reply. Have been a bit busy this week with a new project at work. 

> To make sure we're capturing any other unknown behaviors we might want to directly copy how ActiveRecord does it in find and find_by, which both delegate to find_by_ids. Looks like it's mostly the same as what we have here, but behaves differently in some edge cases -- e.g. it uses .kind_of?(Array) which allows subclasses where .class == Array doesn't.

That seems like a good approach. Looking closely, it appears that only ActiveRecord's `find` delegates to `find_with_ids`:

```ruby
def find(*args)
  return super if block_given?
  find_with_ids(*args)
end
```
[Link](https://github.com/rails/rails/blob/fd601e28d6f98780ab28263a379e98b9a3a92cbd/activerecord/lib/active_record/relation/finder_methods.rb#L67-L70)

It appears that ActiveRecord's `find_by` uses `where` under the hood:

```ruby
def find_by(arg, *args)
  where(arg, *args).take
end
```
[Link](https://github.com/rails/rails/blob/fd601e28d6f98780ab28263a379e98b9a3a92cbd/activerecord/lib/active_record/relation/finder_methods.rb#L80-L82)

In that case, do you recommend leaving `friendly_uuid`'s implementation of `find_by` as-is and updating `expand` to directly copy ActiveRecord's implementation of `find_with_ids`? Or do you think it would be a better idea to basically replace `friendly_uuid`'s `find`'s `super` with the body of ActiveRecord's `find_with_ids`? Or perhaps there's another way that I'm missing?

Reply:
Ah I must have misread then, thanks for the correction.

This PR is in pretty good shape and I'd hate to delay it any longer just for small edge cases and abstract improvements. If you're comfortable with it, I'd love to merge this in its current state, and relegate this discussion to a different PR if you want to continue it. In that PR it may be that overriding `where` would be the right choice.

But the original goal of this PR seems accomplished 👍 If you agree, I will go ahead and merge it!

Reply:
Not a problem!

Sure, I'm fine with merging this PR in as is and creating a new issue for edge cases, etc. Thanks!